### ISSUE TITLE
Using dynamo mapped object type property attributes

### ISSUE THREAD

**Issue by tyronedougherty** [gadelkareem/sails-dynamodb#27]
Hi, does anyone know how to use attributes that are Maps in DynamoDB with waterline?

The JSON type doesn't appear to work 😕 

Many thanks to anyone who may be able to help!

**Comment by tyronedougherty** [comment]
Correct me if I'm wrong but after much digging, it appears that JSON types are not supported?

Line 1126 of index.js (_setColumnType):

```
case "array":  // not support
          schema.StringSet(name, options);
          break;

             //case "json":
             //case "string":
             //case "binary":
        case "string":
             ....
```

Is there any chance of supporting JSON attributes in the near future?

**Comment by devinivy** [comment]
Definitely would take a PR!

**Comment by tyronedougherty** [comment]
Thanks for clarifying @devinivy, in my quick assessment it seems like it may be a significant update and may require updating the vogels package (which is currently very out of date).

**Comment by devinivy** [comment]
Yeah, this module needs a makeover.  Some day I'll have the time to do it, but for the time being I can just support by reviewing PRs.
